scripts/macos-instructions.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+# This script requires yq (https://github.com/mikefarah/yq) which can be installed with this command
+# snap install yq
+
+# This script does two things
+# 1. Modify README.md to include the instructions specified in $INSTRUCTION_FILE
+# 2. Create a github action running all of the code specified in $INSTRUCTION_FILE
+
+# This script must be run in the directory dockstore/dockstore
+
+INSTRUCTION_FILE=macos_instructions.yml
+OUTPUT_FILE_TEXT=macos-instructions.md
+ACTION_FILE=.github/workflows/macos_installation_instructions.yml
+README=README.md
+
+# Initalize output file
+> "$OUTPUT_FILE_TEXT"
+
+NUMBER_OF_INSTRUCTIONS=$(yq '.setupInformation | length' "$INSTRUCTION_FILE")
+
+
+# Add information in actionOnlyInformation to $ACTION_FILE
+yq '.actionOnlyInformation' "$INSTRUCTION_FILE" > "$ACTION_FILE"
+
+
+export i=0
+export CODE_FOR_ACTION=""
+for (( i=0; i<"$NUMBER_OF_INSTRUCTIONS"; i++ ))
```

Comment:
don't do this now, I think this is already good "enough" for the mac os. 
That said, without looking at the details of this particular script, this kinda feels like a job for https://mustache.github.io/mustache.5.html which we've used before. 
Do spin off a ticket and/or explain what the differences might be if mustache cannot simplify this

Reply:
Just having a quick look through the mustache documentation, I think it might work for generating `README.md` but I am unsure about `.github/workflows/macos_installation_instructions.yml`. 

As mustache seems suited to just filling out templates. However, `scripts/macos-instructions.sh` copies all of the elements inside of the `code` key into `.github/workflows/macos_installation_instructions.yml`. Meaning that if you wanted to add an extra key to a command, you could do so without having to alter the script. But, I don't know if this would be possible with mustache. 

I have created a ticket for this [SEAB-4680](https://ucsc-cgl.atlassian.net/browse/SEAB-4680).

Reply:
Yeah, I see. You would potentially have to put the commands (currently in macos_instructions.yml) into a variables file (would look a bit like https://github.com/dockstore/compose_setup/blob/develop/dockstore_launcher_config/compose.config ) but would also need to support loops for a variable number of steps.  The variables file wouldn't be particularly readable and would template generate the readme and github action. 

It isn't immediately clear whether the simplification in re-using mustache (already used in other repos) would be worth it, so I think we can leave it as a ticket,